### PR TITLE
Fix failed test case of internal/safety package

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:

This PR fix the failed test case in internal/safety package.

### Environment:

- Go Version: 1.14.3
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.11.5

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/contributing/coding-style.md [hunk 1]
```diff
@@ -509,3 +509,87 @@ for name, fn := range tests {
 }
 
 ```
+
+### Generate test code
+
+We implement our own [gotests](https://github.com/cweill/gotests) template to generate test code. If you want to install `gotest` tools please execute the following command under the project root directory.
```
Comment (hlts2) [inline docs/contributing/coding-style.md:515]:
```suggestion
We implement our own [gotests](https://github.com/cweill/gotests) template to generate test code. If you want to install `gotest` tools. Please execute the following command under the project root directory.
```
🤔

File: docs/contributing/coding-style.md [hunk 2]
```diff
@@ -509,3 +509,87 @@ for name, fn := range tests {
...
+func init() {
+    log.Init()
+}
+```
+
+and place it on the header of the test file.
+
+2. goleak option
+
+By default, the generated test code will use [goleak](https://github.com/uber-go/goleak) library to test if there is any Goroutine leak. Sometimes you may want to skip the detection, for example, Vald uses [fastime](https://github.com/kpango/fastime) library but the internal Goroutine is not closed due to the needs of the library. To skip the goleak detection we need need to create the following variable to store the ignore function.
```
Comment (hlts2) [inline docs/contributing/coding-style.md:548]:
```suggestion
By default, the generated test code will use [goleak](https://github.com/uber-go/goleak) library to test if there is any Goroutine leak. Sometimes you may want to skip the detection, for example, Vald uses [fastime](https://github.com/kpango/fastime) library but the internal Goroutine is not closed due to the needs of the library. To skip the goleak detection we need to create the following variable to store the ignore function.
```

File: docs/contributing/coding-style.md [hunk 3]
```diff
@@ -509,3 +509,87 @@ for name, fn := range tests {
 }
 
 ```
+
+### Generate test code
+
+We implement our own [gotests](https://github.com/cweill/gotests) template to generate test code. If you want to install `gotest` tools. Please execute the following command under the project root directory.
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply docs/contributing/coding-style.md:515]: Sorry I think I made a mistake.
```suggestion
We implement our own [gotests](https://github.com/cweill/gotests) template to generate test code.
If you want to install `gotest` tools, please execute the following command under the project root directory.
```

File: docs/contributing/coding-style.md [hunk 4]
```diff
@@ -509,3 +509,87 @@ for name, fn := range tests {
...
+
+```golang
+func init() {
+    log.Init()
+}
+```
+
+and place it on the header of the test file.
+
+2. goleak option
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply docs/contributing/coding-style.md:546]: Seems it cannot auto-numbering the list... I will try it later again

File: docs/contributing/coding-style.md [hunk 5]
```diff
@@ -509,3 +509,87 @@ for name, fn := range tests {
...
+        defer goleak.VerifyNone(tt)
+
+// after
+for _, test := range tests {
+    t.Run(test.name, func(tt *testing.T) {
+        // modify the following line
+        defer goleak.VerifyNone(tt, goleakIgnoreOptions...)
+```
+
+3. Defer function
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply docs/contributing/coding-style.md:574]: same as above

**Comment by kevindiu** [comment]
@kpango This PR implements the test case of `internal/safety` test cases. 
Since this PR customized the testing code by adding the `defer()` function to test the `recover()` result. 
I think the similar test case may exists on other packages, I think it is better for you to review this changes. 

I added the following code to the test code:
Line 27-36: https://github.com/vdaas/vald/pull/464/files#diff-d24e608c08457561fd4b754590a38847R27-R36
Line 44: https://github.com/vdaas/vald/pull/464/files#diff-d24e608c08457561fd4b754590a38847R44
Line 131-144: https://github.com/vdaas/vald/pull/464/files#diff-d24e608c08457561fd4b754590a38847R131-R144

It is not urgent but can you please review it in this week (6/19)?

**Pull request merged**